mdal/frmts/mdal_flo2d.cpp

```diff
@@ -675,3 +675,171 @@ std::unique_ptr< MDAL::Mesh > MDAL::DriverFlo2D::load( const std::string &result
+
+void MDAL::DriverFlo2D::addToHDF5File( DatasetGroup *group )
+{
+  saveNewHDF5File( group );
+}
+
+void MDAL::DriverFlo2D::saveNewHDF5File( DatasetGroup *group )
+{
+  // Create file
+  HdfFile file( group->uri(), true );
+
+
+
+  // Create dataspace for dataset File Version
+  std::vector<hsize_t> dimsSingle = {1};
+  float versionValue[] = {99.99f};
+  std::vector<hsize_t> dimsDouble = {1, 1};
+  HdfDataspace dscFileVersion( dimsSingle, true );
+
+  // Create float dataset File Version
+  HdfDataset dsFileVersion( file.id(), "/File Version", true );
+  dsFileVersion.writeFloatArray( dscFileVersion.id(), versionValue );
+
+  // Create dataspace for dataset File Type
+  HdfDataspace dscFileType( dimsSingle, true );
+
+  // Create string dataset File Type
+  HdfDataset dsFileType( file.id(), "/File Type", true );
+  dsFileType.writeString( file.id(), dscFileType.id(), "Xmdf" );
+
+  // Create group TIMDEP NETCDF OUTPUT RESULTS
+  HdfGroup groupTNOR( file.id(), "/TIMDEP NETCDF OUTPUT RESULTS", true );
+
+  // Crete dataspace for attribute
+  HdfDataspace dscTNOR( dimsSingle, true );
+
+  // Create attribute
+  HdfAttribute attTNORGrouptype( groupTNOR.id(), "Grouptype", true );
+  // Write string value to attribute
+  attTNORGrouptype.writeString( dscTNOR.id(), "Generic" );
+
+  for(auto dsGroup : group->mesh()->datasetGroups)
+  {
+    std::string dsGroupName = dsGroup->name();
+    if (dsGroupName == "Bed Elevation" || dsGroupName == "scalarGrp")
+        continue;
+
+    const size_t timesCount = dsGroup->datasets.size();
+    std::vector<hsize_t> timesCountVec;
+
+    std::vector<hsize_t> dimsForValues = { timesCount, dsGroup->mesh()->facesCount() };
+
+    timesCountVec.push_back(static_cast<hsize_t>(timesCount));
+    double max = dsGroup->statistics().maximum;
+    double min = dsGroup->statistics().minimum;
+
+    std::vector<float> maximums;
+    std::vector<float> minimums;
+    std::vector<double> times;
+
+    const size_t valuesSize = timesCount * dsGroup->mesh()->facesCount();
+    std::vector<float> values ( valuesSize );
+
+    size_t valuesIndex = 0;
+    if (dsGroupName != "Velocity")
+    {
+        for (size_t i = 0; i < dsGroup->datasets.size(); i++)
+        {
+            const size_t singleValuesCount = dsGroup->datasets[i]->valuesCount();
+
+            std::unique_ptr<double[]> rawValueData(new double[singleValuesCount]);
+            dsGroup->datasets[i]->scalarData(0, singleValuesCount, rawValueData.get());
+
+            for (size_t j = 0; j < singleValuesCount; j++)
+            {
+                values.at(valuesIndex++) = static_cast<float>(rawValueData[i]);
+            }
+        }
+    }
+
+    for( size_t i = 0; i < timesCount; i++)
+    {
+        maximums.push_back(static_cast<float>(max));
+        minimums.push_back(static_cast<float>(min));
+        times.push_back(dsGroup->datasets[i]->time());
+    }
+
+    HdfGroup groupFlowDepth( groupTNOR.id(), "/TIMDEP NETCDF OUTPUT RESULTS/" + dsGroupName, true );
```

Comment:
create 2 or 4 subfunctions: `store[Scalar/Vector][Face/Vertex]DatasetGroup` . one function should do 1 action

Reply:
Agree, will do the in final version. Keeping it this way for now for simpler development and moving things around.